packages/ui/src/multiplying-architecture/KaotoEditorChannelApi.ts

```diff
@@ -57,4 +57,9 @@ export interface KaotoEditorChannelApi extends KogitoEditorChannelApi {
     exclude?: string,
     options?: Record<string, unknown>,
   ): Promise<string[] | string | undefined>;
+
+  /**
+   * Detect if the current opened file is part of a Maven project or not
+   */
+  isInsideMavenProject(): Promise<boolean>;
```

Comment:
How should Kaoto use this? What should be enabled or disabled based on this information?

Reply:
I implemented the functionality from the KTO-503. I have not looked to which functionality it will enable. But first ida is to not call to get the Maven runtime information if it is not a Maven project. Or when it is a Maven project but the runtime information returned is undefined alert the user that we were unable to retrieve them.